src/server/dev.ts

```diff
@@ -149,6 +149,10 @@ export async function createDevServer(
         _loadedEntry.default ||
         _loadedEntry;
 
+      if (_handler.options && _handler.options.onError) {
+        app.options.onError = _handler.options.onError;
```

Comment:
I was thinking we could inherit/merge all options. For example for this https://github.com/unjs/h3/issues/693 (and `debug` option)

Reply:
Of course this would be better, but I've tested how to do the merge and some of the hooks trigger twice

for example ↓

```ts
if (_handler.options) {
  // This causes some hooks to run twice
  app.options = Object.assign(app.options, _handler.options)
}
```
I think this might be a bug in h3, but I think I'd have to look deeper at h3 as well.

I like https://github.com/unjs/h3/issues/693，But I'd like to see the logic of your implementation, let's try to keep the behavior consistent 🥰